src/Main.cpp

```diff
@@ -110,45 +109,42 @@ class ConfigCB : public Config::Callback {
 ConfigCB::ConfigCB(Main &main) : main(main) {}
 
 void ConfigCB::operator()(std::string const &msg) {
-  using std::string;
-  using namespace rapidjson;
+  using nlohmann::json;
   LOG(7, "Command received: {}", msg.c_str());
-  Document j0;
-  j0.Parse(msg.c_str());
-  if (j0.HasParseError()) {
-    LOG(3, "Command does not look like valid json");
-    return;
-  }
-  auto m1 = j0.FindMember("cmd");
-  if (m1 == j0.MemberEnd()) {
-    return;
-  }
-  if (!m1->value.IsString()) {
-    return;
+  try {
+    auto Document = json::parse(msg);
+    handleParsedJSON(Document);
+  } catch (...) {
+    LOG(3, "Command does not look like valid json: {}", msg);
   }
-  string cmd = m1->value.GetString();
-  if (cmd == "add") {
-    auto m2 = j0.FindMember("streams");
-    if (m2 == j0.MemberEnd()) {
-      return;
+}
+
+void ConfigCB::handleParsedJSON(nlohmann::json const &Document) {
+  using std::string;
+  using nlohmann::json;
+  if (auto x = find<string>("cmd", Document)) {
+    auto Command = x.inner();
+    if (Command == "add") {
+      if (auto x = find<json>("streams", Document)) {
```

Comment:
I'm not keen on using `x` here at all, but it should definitely be avoided when there is another `x` in the upper scope.